comp/core/configsync/configsyncimpl/module.go

```diff
@@ -100,17 +93,20 @@ func newConfigSync(deps dependencies, agentIPCPort int, configRefreshIntervalSec
 		ctx:       ctx,
 	}
 
-	if deps.SyncParams.OnInit {
-		if deps.SyncParams.Delay != 0 {
-			select {
-			case <-ctx.Done(): //context cancelled
-				// TODO: this component should return an error
+	if deps.SyncParams.OnInitSync {
+		deps.Log.Infof("triggering configsync on init (will retry for %s)", deps.SyncParams.OnInitSyncTimeout)
+		deadline := time.Now().Add(deps.SyncParams.OnInitSyncTimeout)
+		for {
+			if err := configSync.updater(); err == nil {
+				break
+			}
+			if time.Now().After(deadline) {
 				cancel()
-				return nil
-			case <-time.After(deps.SyncParams.Delay):
+				return nil, deps.Log.Errorf("failed to sync config at startup, is the core agent listening on '%s' ?", url.String())
```

Comment:
Just checking whether it's intentional, but if this error is returned due to all retries failing, that will cause the app to exit due to fx graph construction failing.

Reply:
Yes, I think starting the process in a failed state is no OK. Especially with k8s which might restart it and fix the error.

Reply:
Yup, I think that makes sense.